bcipy/helpers/task.py

```diff
@@ -370,186 +373,153 @@ def get_key_press(
     return None
 
 
+def data_reshaper(input_data_type: InputDataType, *args, **kwargs):
+    if input_data_type == InputDataType.TRIAL:
+        return trial_reshaper(*args, **kwargs)
+    elif input_data_type == InputDataType.INQUIRY:
+        return inquiry_reshaper(*args, **kwargs)
+    else:
+        raise NotImplementedError(f"Unsupported input data type: {input_data_type}")
+
+
+def grouper(iterable, n, fillvalue=None):
+    args = [iter(iterable)] * n
+    return zip_longest(*args, fillvalue=fillvalue)
+
+
+def inquiry_reshaper(trial_target_info: list,
+                     timing_info: list,
+                     eeg_data: np.ndarray,
+                     fs: int,
+                     trials_per_inquiry: int,
+                     offset: float = 0,
+                     channel_map: List[int] = DEFAULT_CHANNEL_MAP,
+                     trial_length: float = 0.5):
+    """Extract inquiry data and labels.
+    Args:
+        trial_target_info (list): labels each trial as "target", "non-target", "first_pres_target", etc
+        timing_info (list): Timestamp of each event in seconds
+        eeg_data (np.ndarray): shape (channels, samples) preprocessed EEG data
+        fs (int): sample rate of preprocessed EEG data
+        trials_per_inquiry (int): number of trials in each inquiry
+        offset (float, optional): Any calculated or hypothesized offsets in timings. Defaults to 0.
+        channel_map (tuple, optional): Describes which channels to include or discard. Defaults to DEFAULT_CHANNEL_MAP.
+        trial_length (float, optional): [description]. Defaults to 0.5.
+
+    Returns:
+        reshaped_data (np.ndarray): inquiry data of shape (Channels, Inquiries, Samples)
+        labels (np.ndarray): integer label for each inquiry. With trials_per_inquiry=K,
+            a label of [0, K-1] indicates position of "target", or label of K indicates
+            target was not present.
+    """
+    # Remove the channels that we are not interested in
+    channels_to_remove = [idx for idx, value in enumerate(channel_map) if value == 0]
+    eeg_data = np.delete(eeg_data, channels_to_remove, axis=0)
+
+    # Number of samples we are interested per inquiry
+    num_samples = int(trial_length * fs * trials_per_inquiry)
+
+    # Remove unwanted elements from target info and timing info
+    targets_included = set(["target", "nontarget"])
```

Comment:
We should ask for the triggers we want to be included/excluded. This could be used for label assignments as well.

Reply:
Ok - I added these with the defaults being `included = set(["target", "nontarget"])` and `excluded = set([])`.

Since these sets are not mutated within the function, I think it's OK (and keeps the code simple) to just construct them in the function args.